### PR TITLE
feat(tooling) upgrade tooling to recent versions

### PR DESCRIPTION
**Pull request by dduportal**
This introduces the following tooling changes:

- Fix a bug in the github workflo to normalize the Docker cache image name
- Bump nodeJS to last LTS (v16)
- Ensure that the Dockerfile passes the hadolint linter
- Allow the command line `ncu` to be executed from the working dir in the container (with the `make shell` command) and works as expected
- Simplifies the `Makefile` and `docker-compose.yml` 
- Bump the PDF generator to last available stable version
- Disable the htmlproofer system
- Enable RevealJS plugins to provide a Table of Content menu, toolbar, copy and paste of code blocks, etc.
- Update qrcode generation system to fix some mobile phone cameras
- Bump all NPm dependencies to their latest except `asciidoctor/revealjs` and plugin `revealjs-menu`
- Update example content to demonstrate some content improvements/changes

### CODE REVIEW

**Comment by dduportal** [comment]
Ready to roll!

File: .github/workflows/build-workflow.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ on:
 
 env:
   ## Override default value for Docker cached image
-  IMAGE_CACHE_NAME: "ghcr.io/${{ github.repository_owner }}/slides"
+  IMAGE_CACHE_NAME_ORIGINAL: "ghcr.io/${{ github.repository_owner }}/slides:cicd-lectures"
```
Comment (jmMeessen) [inline .github/workflows/build-workflow.yml:17]: I guess that `cicd-lectures` is harmless (it is just a cache) or I didn't understand how it works

File: content/attributes.adoc [hunk 2]
```diff
@@ -17,10 +17,8 @@
 :source-highlighter: highlightjs
 :highlightjs-main-theme: github
 :highlightjs-theme: {revealjsdir}/plugin/highlight/styles/{highlightjs-main-theme}.min.css
-:invert: state=invert,background-color="white"
-// Enable font awesome, and use it from local resources (instead of CDN)
-:icons: font
+:invert: state=invert,background-color="rgb(248, 228, 130)"
```
Comment (jmMeessen) [inline content/attributes.adoc:20]: I made this evolve in the current templates: `:invert: state=invert,background-color="BLANCHEDALMOND"`

will have to crosscheck that when rebasing

File: content/attributes.adoc [hunk 3]
```diff
@@ -2,7 +2,7 @@
 :allow-uri-read:
 :linkattrs:
 :idprefix:
-:imagesdir: images
+:imagesdir: media/
```
Comment (jmMeessen) [inline content/attributes.adoc:5]: Will have to see the impact of the working branch as it uses `images`. Sources will have to be changed (not too difficult).

File: content/attributes.adoc [hunk 4]
```diff
@@ -17,10 +17,8 @@
 :source-highlighter: highlightjs
 :highlightjs-main-theme: github
 :highlightjs-theme: {revealjsdir}/plugin/highlight/styles/{highlightjs-main-theme}.min.css
-:invert: state=invert,background-color="white"
-// Enable font awesome, and use it from local resources (instead of CDN)
```
Comment (jmMeessen) [inline content/attributes.adoc:21]: These lines were removed too in my working branch. The changes should be retrofitted in Master

**Review by jmMeessen — COMMENTED** [review]
We can merge it tomorrow together. (it is late)

Thank you very much for the PR.

File: .github/workflows/build-workflow.yml (hunk 1 above)
Comment (dduportal) [reply .github/workflows/build-workflow.yml]: it was a mistake. Should be fixed.

File: content/attributes.adoc (hunk 2 above)
Comment (dduportal) [reply content/attributes.adoc:20]: should not be an issue when rebasing (worst case a conflict to solve once)

File: content/attributes.adoc (hunk 3 above)
Comment (dduportal) [reply content/attributes.adoc:5]: Yes, this change is, alas, mandatory for the "stream" system used in gulp, to author media files such as videos, qrcode, etc.

File: content/attributes.adoc (hunk 4 above)
Comment (dduportal) [reply content/attributes.adoc:21]: rebase should detect this. Let's see.

**Pull request merged**